### ISSUE TITLE
Folder extraction does not work

### ISSUE THREAD

**Issue by Skyleiger** [srikanth-lingala/zip4j#348]
Hi,
I'd like to extract a folder from my zip file, but it doesn't work.
I got the following exception:
```
[2021-08-13 07:00:51.489         Console d.d.c.c.ConsoleSender                     INFO]:   net.lingala.zip4j.exception.ZipException: File header and local file header mismatch
[2021-08-13 07:00:51.489         Console d.d.c.c.ConsoleSender                     INFO]:       at net.lingala.zip4j.tasks.AbstractExtractFileTask.verifyNextEntry(AbstractExtractFileTask.java:151)
[2021-08-13 07:00:51.489         Console d.d.c.c.ConsoleSender                     INFO]:       at net.lingala.zip4j.tasks.AbstractExtractFileTask.extractFile(AbstractExtractFileTask.java:59)
[2021-08-13 07:00:51.490         Console d.d.c.c.ConsoleSender                     INFO]:       at net.lingala.zip4j.tasks.ExtractFileTask.executeTask(ExtractFileTask.java:43)
[2021-08-13 07:00:51.490         Console d.d.c.c.ConsoleSender                     INFO]:       at net.lingala.zip4j.tasks.ExtractFileTask.executeTask(ExtractFileTask.java:22)
[2021-08-13 07:00:51.490         Console d.d.c.c.ConsoleSender                     INFO]:       at net.lingala.zip4j.tasks.AsyncZipTask.performTaskWithErrorHandling(AsyncZipTask.java:51)
[2021-08-13 07:00:51.490         Console d.d.c.c.ConsoleSender                     INFO]:       at net.lingala.zip4j.tasks.AsyncZipTask.access$400(AsyncZipTask.java:9)
[2021-08-13 07:00:51.491         Console d.d.c.c.ConsoleSender                     INFO]:       at net.lingala.zip4j.tasks.AsyncZipTask$1.run(AsyncZipTask.java:36)
[2021-08-13 07:00:51.491         Console d.d.c.c.ConsoleSender                     INFO]:       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2021-08-13 07:00:51.491         Console d.d.c.c.ConsoleSender                     INFO]:       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2021-08-13 07:00:51.491         Console d.d.c.c.ConsoleSender                     INFO]:       at java.base/java.lang.Thread.run(Thread.java:829)
```

I use the following code to extract the folder: 
```
ZipFile zipFile = new ZipFile(archive.toFile());
zipFile.setRunInThread(true);
zipFile.extractFile("data/", "/srv/test/");
```

Thank you for your help.
Dominic

**Comment by Skyleiger** [comment]
I'm using the latest version: 2.9.0
I've attached the zip file.
[05fb50b0-7be9-4d1c-b137-9bf3486a5465.zip](https://github.com/srikanth-lingala/zip4j/files/7079588/05fb50b0-7be9-4d1c-b137-9bf3486a5465.zip)

**Comment by srikanth-lingala** [comment]
Thanks for the zip file. I am able to reproduce the issue. Will look into this.

**Comment by srikanth-lingala** [comment]
Fixed in v2.9.1 released today.
